client/src/contexts/FeedbackContext.tsx

```diff
@@ -1,49 +1,46 @@
 import { getFromStorage, setInStorage } from 'API/localStorage';
-import { FeedbackData } from 'components/Data/FeedbackData';
-import React, { createContext, FC, useState, useMemo, useCallback, useEffect } from 'react';
+import { mockFeedbackData } from 'components/Data/mockFeedbackData';
+import React, { createContext, FC, useState, useCallback, useEffect } from 'react';
 import { Feedback } from 'types';
 
 type FeedbackContextData = {
   feedback: Feedback[];
-  setFeedback: (feedback: Feedback[]) => void;
   addFeedback: (newFeedback: Feedback) => void;
 };
 
 const FeedbackContext = createContext<FeedbackContextData>({
   feedback: [],
-  setFeedback: () => {
-    return null;
-  },
   addFeedback: () => {
     return null;
   }
 });
 
+const LOCALSTORAGE_LABEL = 'feedback';
+
 export const FeedbackProvider: FC<React.ReactNode> = ({ children }) => {
   const [feedback, setFeedback] = useState<Feedback[]>([]);
 
   const fetchFeedback = useCallback(() => {
-    const feedbackData = getFromStorage('feedback');
-    if (feedbackData) {
-      setFeedback(feedbackData);
-    } else {
-      setInStorage('feedback', FeedbackData);
-      setFeedback(FeedbackData);
-    }
-  }, [setFeedback]);
+    const feedbackData = getFromStorage(LOCALSTORAGE_LABEL);
+    setFeedback(feedbackData || mockFeedbackData);
+    if (!feedbackData) setInStorage(LOCALSTORAGE_LABEL, mockFeedbackData);
+  }, []);
 
   useEffect(() => {
     fetchFeedback();
   }, [fetchFeedback]);
 
-  const feedbackProvider = useMemo(() => {
-    const addFeedback = (newFeedback: Feedback) => {
-      setInStorage('feedback', [newFeedback, ...feedback]);
-    };
-    return { feedback, setFeedback, addFeedback };
-  }, [feedback]);
-
-  return <FeedbackContext.Provider value={feedbackProvider}>{children}</FeedbackContext.Provider>;
+  const addFeedback = (newFeedback: Feedback) => {
+    setInStorage(LOCALSTORAGE_LABEL, [newFeedback, ...feedback]);
+    setFeedback([newFeedback, ...feedback]);
+  };
+  const feedbackProviderValue = () => {
+    return { feedback, addFeedback };
+  };
```

Comment:
why do you need function here?

Reply:
To avoid useMemo :)

Reply:
I get this error : The 'feedbackProviderValue' object (at line 37) passed as the value prop to the Context provider (at line 40) changes every render. To fix this consider wrapping it in a useMemo hook.eslint[react/jsx-no-constructed-context-values]

Reply:
ok, you are trying to avoid error. please explain me what way this function will be better than plain variable from any perspective?

Reply:
Nothing better, I just get eslint error.
![Screenshot 2022-09-23 141632](https://user-images.githubusercontent.com/55662628/191949273-8f2f3d8a-1366-43d5-8be9-d0b184a1c933.png)

Reply:
what would you suggest to do?

Reply:
eslint error is not a reason, i can add to your config such a rules that will show errors for every code line. you are a  code owner, please explain differences between these two solutions as react developer

Reply:
The first one declares function while the second one declares a simple object. In JavaScript, functions are also objects, so there is almost no difference between the two except that in the first example you can call as a function. I think, those two approaches are pretty much exactly the same. A function instance is just an object . But, generally this is not how a function is used. A function is a procedure to be performed to achieve something. So I would use the second example and set to off eslint  ‘react/jsx-no-constructed-context-values’.

Reply:
as far as you say that these two solutions have no difference and eslint doesn't like one of these solutions - that means second one also won't be the best choice according to exactly this eslint rule. and all that means you have 2 options here:
1. remove this rule and fix your code
2. fix your code according to eslint rule